src/StatementBase.php

```diff
@@ -60,7 +60,7 @@ public function __construct() {
 
     private function _asVersion(&$result, $version) {
         foreach ($result as $property => $value) {
-            if ($value !== false && empty($value)) {
```

Comment:
Empty is problematic because it is too broad. See the list of return values here: http://php.net/manual/en/function.empty.php